File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/clips/ClipsViewModel.java
package com.exact.twitch.ui.clips;


import androidx.lifecycle.MutableLiveData;
import androidx.annotation.Nullable;

import com.exact.twitch.model.clip.Clip;
import com.exact.twitch.repository.TwitchService;
import com.exact.twitch.ui.viewmodel.PagedListViewModel;

import javax.inject.Inject;

public class ClipsViewModel extends PagedListViewModel<Clip> {

    private MutableLiveData<CharSequence> sortText = new MutableLiveData<>();
    private Integer selectedSortItem;
    private String period;
    private boolean trending;

    @Inject
    ClipsViewModel(TwitchService repository) {
        super(repository);
    }

    public void loadClips(@Nullable String channelName, @Nullable String gameName, @Nullable String languages, @Nullable String period, @Nullable Boolean trending, boolean reload) {
        loadData(repository.loadClips(channelName, gameName, languages, period, trending), reload);
    }

    public void loadFollowedClips(String userToken, @Nullable Boolean trending, boolean reload) {
        loadData(repository.loadFollowedClips(userToken, trending), reload);
    }

    public MutableLiveData<CharSequence> getSortText() {
        return sortText;
    }

    public Integer getSelectedSortItem() {
        return selectedSortItem;
    }

    public void setSelectedSortItem(Integer selectedSortItem) {
        this.selectedSortItem = selectedSortItem;
    }

    public boolean getTrending() {
        return trending;
    }

    public void setTrending(boolean trending) {
        this.trending = trending;
    }

    public String getPeriod() {
        if (period == null) {
            period = "week";
        }
        return period;
    }

    public void setPeriod(String period) {
        this.period = period;
    }
}

File: app/src/main/java/com/exact/twitch/ui/clips/ClipsViewModel.kt
package com.exact.twitch.ui.clips

import androidx.lifecycle.MutableLiveData
import com.exact.twitch.model.clip.Clip
import com.exact.twitch.repository.TwitchService
import com.exact.twitch.ui.Initializable
import com.exact.twitch.ui.common.PagedListViewModel
import javax.inject.Inject

class ClipsViewModel @Inject constructor(
        private val repository: TwitchService) : PagedListViewModel<Clip>(), Initializable {

    val sortText = MutableLiveData<CharSequence>()
    var period: Period? = null
    var trending: Boolean = false

    fun loadClips(channelName: String? = null, gameName: String? = null, languages: String? = null, reload: Boolean) {
        loadData(repository.loadClips(channelName, gameName, languages, period ?: Period.WEEK, trending, compositeDisposable), reload)
    }

    fun loadFollowedClips(userToken: String, reload: Boolean) {
        loadData(repository.loadFollowedClips(userToken, trending, compositeDisposable), reload)
    }

    override fun isInitialized(): Boolean {
        return sortText.value == null
    }
}
